File: 프로그래머스/3/70130. 스타 수열/스타 수열.py
def solution(a):
    hash = dict()
    answer = 0
    stack = []
    if len(a) == 1 :
        return 0
    elif len(a) == 2 :
        if a[0] != a[1] :
            return 1
        else :
            return 0
    #맨앞에 두개 같으면 하나 안셈
    if a[0] != a[1] :
        hash[a[0]] = 1
        
   
    if a[-1] != a[-2]:
        hash[a[-1]] = 1
            
    for i in range(1,len(a) - 1):
        if not(a[i-1] == a[i] and a[i] == a[i + 1]):
            if a[i] in hash :
                hash[a[i]] += 1
            else :
                hash[a[i]] = 1
                
    Key = -1
    count = 0
    for i in hash :
        if hash[i] > count :
            count = hash[i]
            Key = i
    #print(Key)
    for i in a :
        if i == Key :
            if stack :
                if stack[-1] != Key :
                    stack.clear()
                    answer += 1
            else :
                stack.append(i)
        else :
            if stack :
                if stack[-1] == Key :
                    stack.clear()
                    answer += 1
            else :
                stack.append(i)
    return answer * 2

File: 프로그래머스/3/70130. 스타 수열/스타 수열.java
import java.util.*;
class Solution {
    public int solution(int[] a) {
        int answer = 0;
        int N = a.length;
        if(N == 1) return 0;
        
        Map<Integer,Integer> map = new HashMap<>();
        
        // 1. 각 원소들의 개수를 센다
        for(int i = 0; i < N; i++){
            if(!map.containsKey(a[i]))
                map.put(a[i],1);
            else 
                map.put(a[i],map.get(a[i]) + 1);
        }
        if(map.keySet().size() == 1) return 0;
        
        for(Integer key : map.keySet()){
            if(map.get(key) <= answer) continue;
            int count = 0;
            
            for(int i = 0; i < N - 1; i++){
                if(a[i] != key && a[i + 1] != key) continue;
                if(a[i] == a[i + 1]) continue;
                count++;
                i++;
            }
            answer = Math.max(answer,count);
        }
        
        return answer * 2;
    }
}